lib/components/modal.vue

```diff
@@ -82,14 +82,28 @@
     import bBtn from './button.vue';
 
     const FOCUS_SELECTOR = [
-        'button:not([disabled]):not([style*="display: none"]):not([style*="display:none"])',
-        'input:not([disabled]):not([style*="display: none"]):not([style*="display:none"])',
-        'select:not([disabled]):not([style*="display: none"]):not([style*="display:none"])',
-        'textarea:not([disabled]):not([style*="display: none"]):not([style*="display:none"])',
-        'a:not([disabled]):not(.disabled):not([style*="display: none"]):not([style*="display:none"])',
-        '[tabindex]:not([disabled]):not(.disabled):not([style*="display: none"]):not([style*="display:none"])'
+        'button:not([disabled])',
+        'input:not([disabled])',
+        'select:not([disabled])',
+        'textarea:not([disabled])',
+        'a:not([disabled]):not(.disabled)',
+        '[tabindex]:not([disabled]):not(.disabled)'
     ].join(',');
 
+    // Determine if an HTML element is visible - Faster than CSS check
+    function isVisible(el) {
+        return el && (el.offsetWidth > 0 || el.offsetHeight > 0);
+    }
+    
+    // Find the first visible element contained in a given root element
+    function findFirstVisible(root, selector) {
+        if (!root || !root.querySelectorAll || !selector) {
+            return null;
+        }
+        let els = Array.prototype.slice.call(root.querySelectorAll(selector));
```

Comment:
Although i could check for the native function first, and then fallback to the simple while loop